### ISSUE TITLE
Not compatible with nodebb 1.0.0

### ISSUE THREAD

**Issue by Nisthar** [ahwayakchih/openshift-nodebb#17]
Hi,

I just tried to upgrade my app and i want to report that its not compatible with nodebb 1.0.0. I got an error `remote: This looks like a new installation, so you'll have to answer a few questions about your environment before we ca
n proceed.
remote: Press enter to accept the default setting (shown in brackets).
remote:
remote: 2/3 23:57 [152461] - info: Now configuring mongo database:
remote: Configuration Saved OK
remote: 2/3 23:57 [152461] - error: Unable to initialize MongoDB! Is MongoDB installed? Error :Cannot find module 'conne
ct-mongo/es5'
remote: 2/3 23:57 [152461] - warn: NodeBB Setup Aborted.
remote:  Error: Cannot find module 'connect-mongo/es5'
remote:     at Function.Module._resolveFilename (module.js:338:15)
remote:     at Function.Module._load (module.js:280:25)
remote:     at Module.require (module.js:364:17)
remote:     at require (module.js:380:17)
remote:     at Object.module.init (/var/lib/openshift/56d3d2e189f5cfad110001a1/app-root/runtime/repo/src/database/mongo.
js:55:20)
remote:     at /var/lib/openshift/56d3d2e189f5cfad110001a1/app-root/runtime/repo/src/install.js:180:25
remote:     at /var/lib/openshift/56d3d2e189f5cfad110001a1/app-root/runtime/repo/src/install.js:555:3
remote:     at Object.oncomplete (fs.js:108:15)`

when the automatic setup is ran after installation.

**Comment by theunknownartisthour** [comment]
Looks like it didn't find the package to handle mongodb? Maybe the update changed to a different package to handle the database. Try sshing in and running npm install.

**Comment by Nisthar** [comment]
@theunknownartisthour Yes. Think so

Good thing i had a snapshot saved. Otherwise, I may have lost of all my users data.

**Comment by ahwayakchih** [comment]
Nodejs cartridge does run `npm install` when new build is being made.

I'll test upgrading to 1.0 next week.

**Comment by ahwayakchih** [comment]
I can confirm that upgrading is broken. I got different errors, but some of them related to lack of `./nodebb upgrade` and some of them caused by bugs in NodeBB 1.x.x.

**Comment by Nisthar** [comment]
@ahwayakchih I thought there were no bugs in their releases. Thought only master branch contains bugs. 
Did this mean that it will take some time to fix this?

**Comment by ahwayakchih** [comment]
@Nisthar some of the bugs are already fixed, but i still see installation freezing, so don't know... I'll try again on or after next weekend and see if it works then.

**Comment by Nisthar** [comment]
@ahwayakchih Ok :+1:

**Comment by ahwayakchih** [comment]
@Nisthar i've tested upgrade once again. Unfortunately it stops with `Error: schema-out-of-date`, even though i installed latest 0.9.x first and then updated to 1.x.x.

Clean 1.x.x installation works fine.

**Comment by kaizhou-vienna** [comment]
@ahwayakchih , first thank you for this comprehensive tutorial, it is definitely a time-saver. I just want to provide the feedback of following it to do a fresh install.
I have tried to do a clean install 1.0.3 with your tutorial. Unfortunately this doesn't work. I am using MongoLab cloud DB and it abort the nodebb setup with msg "Mongo returned the following error: Authentication failed.". I tried with 0.9.x installation, it works only with original Nodejs010 cartridges. The updated nodejs repository in the tutorial will break the installation with msg of port not available. So by now I will keep using old 0.9.x and original nodejs cartridge, wait for the future upgrade.
